### PR TITLE
Add sir-trevor-js v0.6.0-beta9

### LINKED ISSUE #6113: [Request] Sir Trevor
**Library name:** Sir Trevor
**Git repository url:** https://github.com/madebymany/sir-trevor-js.git
**npm package url(optional):** https://www.npmjs.com/package/sir-trevor
**License(s):** MIT
**Official homepage:** http://madebymany.github.io/sir-trevor-js/

### PR DESCRIPTION
**Pull request by x09326**
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): git@github.com:madebymany/sir-trevor-js.git
 * Official website (optional, not the repository): https://madebymany.github.io/sir-trevor-js
 * NPM package url (optional):
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 4231
   - Count of watchers: 160
   - Count of forks: 401
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

with git auto-updated config added close #6113 #6152 cc @raffij

### CODE REVIEW

**Comment by x09326** [comment]
@PeterDaveHello Please help me check it. Thank you.

**Comment by PeterDaveHello** [comment]
@x09326 0.6.0-beta9 is not a stable version, is there any stable version that we can use?

**Comment by x09326** [comment]
@PeterDaveHello 
but the author suggested add 0.6.0-beta9 because if add 0.5 will have a big change in the future.

https://github.com/madebymany/sir-trevor-js/issues/452

**Comment by PeterDaveHello** [comment]
@x09326 Okay, i will take a look at that.

**Comment by raffij** [comment]
@PeterDaveHello 0.6-beta9 is considered the most stable including the latest interface. It's being used in production at present, but there will likely be 1 or 2 more beta version fixing some little issues before pushing the new version to npm.

0.6 is a big release as it removes jQuery and includes a new interface. Therefore the release of this version needs time and to be properly managed, for now we've been pushing people to master or the beta to get the latest 'stable' version.

Hope that helps give a little context into why @x09326 was using a beta version on cdnjs.

**Comment by PeterDaveHello** [comment]
@raffij Thanks for you comments, I got it.

File: ajax/libs/sir-trevor-js/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  "homepage": "https://madebymany.github.io/sir-trevor-js/",
+  "repository": {
+    "type": "git",
+    "url": "git@github.com:madebymany/sir-trevor-js.git"
+  },
+  "keywords": [
+    "content editing",
+    "rich content editor",
+    "Sir Trevor",
+    "SirTrevor"
```
Comment (PeterDaveHello) [inline ajax/libs/sir-trevor-js/package.json]: Line 14, 15 looks the same.

**Comment by PeterDaveHello** [comment]
I found some problems, first @x09326 you didn't add the all `sir-trevor*` matched files as 
https://github.com/madebymany/sir-trevor-js/tree/v0.6.0-beta9, and second, @raffij some files in https://github.com/madebymany/sir-trevor-js/tree/v0.6.0-beta9 looks out-date, e.g. sir-trevor.css.map, it's from v0.5.0, doesn't this matter?

File: ajax/libs/sir-trevor-js/package.json (hunk 1 above)
Comment (raffij) [inline ajax/libs/sir-trevor-js/package.json]: @PeterDaveHello people aren't really sure how to reference the editor so we have both the one word and joined term in there.

**Comment by x09326** [comment]
Are the sir-trevor.test.css and sir-trevor.test.js needed ?

**Comment by raffij** [comment]
test or map files should be in there.

The latest built files are in:
https://github.com/madebymany/sir-trevor-js/tree/b0062d6ade5af3d3cd6b35f18591e35016e2b62a/build

The latest build files will always be:
https://github.com/madebymany/sir-trevor-js/tree/dist/build

There aren't any test and map files in there. Also master isn't always stable. The stable build is the latest release or the dist branch.

File: ajax/libs/sir-trevor-js/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  "dependencies": {
+    "es5-shim": "^4.0.3",
+    "svg4everybody": "^2.0.3"
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/madebymany/sir-trevor-js.git",
+    "basePath": "",
```
Comment (PeterDaveHello) [inline ajax/libs/sir-trevor-js/package.json]: Okay, so @x09326 should set `build` folder as basePath.

**Comment by raffij** [comment]
The only files you'll want in cdnjs are in /build yes.

**Comment by x09326** [comment]
@PeterDaveHello Please help me review it. Thank you.

**Comment by PeterDaveHello** [comment]
@x09326 you didn't remove the old files, I'll recommand you to redo a whole new commit to prevent the wrong files from the previous commit.

**Comment by x09326** [comment]
@PeterDaveHello I couldn't get you. How do I recognize the files are new or old?

**Comment by PeterDaveHello** [comment]
@x09326 you added unrelated files, see:
https://github.com/madebymany/sir-trevor-js/tree/b0062d6ade5af3d3cd6b35f18591e35016e2b62a/build

**Comment by x09326** [comment]
@PeterDaveHello I deleted the test file, and I don't remove the map. Is it right?
Please help me check it, thanks.

**Comment by PeterDaveHello** [comment]
@x09326 So did you make sure all the files came from the tagged version of build folder?

**Comment by x09326** [comment]
Okay, I find the problem. I try to redo it right now.

**Comment by x09326** [comment]
@PeterDaveHello I finished it, I downloaded the latest files and covered the old files.
Please help me review it. Thank you.

**Comment by PeterDaveHello** [comment]
@x09326 The *latest* files? Or the files from 0.6.0-beta9 tag?

**Comment by x09326** [comment]
@PeterDaveHello No, I downloaded from the dist branch.

**Comment by PeterDaveHello** [comment]
@x09326 Why not from 0.6.0-beta9 tag?

**Comment by x09326** [comment]
@PeterDaveHello Because at first I downloaded the 0.6.0-beta a few days ago then I couldn't find the main files, and then I downloaded the master. But the master files are old files.
But now the files I pushed are from dist branch.
@raffij said that dist branch is the latest version. So I downloaded it.

**Comment by PeterDaveHello** [comment]
@x09326 We use tag to recognize the versions, so you should grab the files from the correct tags, not a single branch.

**Comment by x09326** [comment]
Okay. I try redo it again.

**Comment by x09326** [comment]
I check the files in the build folder between tag v0.6.0-beta9 and the dist branch.
They are the same files, So I think it's okay.

**Comment by raffij** [comment]
I'm going to leave you to it as sounds like i've given enough advice now.

**Comment by PeterDaveHello** [comment]
@x09326 Okay, but next time you should grab files from version tag.

**Comment by x09326** [comment]
Got it.
@raffij and @PeterDaveHello, thank you for your help, I learn so mush in this PR.

**Comment by PeterDaveHello** [comment]
@x09326 please add comma in your commit message, and also, you need to put keyword `close` before each issue number, GitHub doesn't support to close many issues by a single `close`.

**Comment by x09326** [comment]
@PeterDaveHello I modified it, please check it. Thank you.

**Comment by PeterDaveHello** [comment]
@x09326 Please send another PR to change filemap to `"**/*"` in a build folder.

**Pull request merged**